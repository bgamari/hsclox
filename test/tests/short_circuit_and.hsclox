if (false and (1==1)) {
    print "then branch";
}
