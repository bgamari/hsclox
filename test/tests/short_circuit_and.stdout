== <script> ==
0000    1 OP_FALSE
0001    | OP_JUMP_IF_FALSE    1 -> 10
0004    | OP_POP
0005    | OP_CONSTANT         0 '1
0007    | OP_CONSTANT         1 '1
0009    | OP_EQUAL
0010    | OP_JUMP_IF_FALSE   10 -> 20
0013    | OP_POP
0014    2 OP_CONSTANT         2 'then branch
0016    | OP_PRINT
0017    3 OP_JUMP            17 -> 21
0020    | OP_POP
0021    4 OP_NIL
0022    | OP_RETURN
